stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/reactions/ReactionsOverlayDialogFragment.kt

```diff
@@ -0,0 +1,161 @@
+        super.onViewCreated(view, savedInstanceState)
+        binding.containerView.setOnClickListener {
+            dismiss()
+        }
+
+        setupEditReactionsView()
+        setupMessageView()
+        setupUserReactionsView()
+        anchorReactionsViewToMessageView()
+    }
+
+    override fun onStart() {
+        super.onStart()
+        dialog?.window?.let {
+            it.setLayout(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT)
+            it.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
+        }
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        _binding = null
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        reactionClickListener = null
+    }
+
+    private fun setupEditReactionsView() {
+        binding.editReactionsView.setMessage(messageItem.message, messageItem.isMine)
+        binding.editReactionsView.setReactionClickListener {
+            reactionClickListener?.onReactionClick(it)
+            dismiss()
+        }
+    }
+
+    /**
+     * TODO: find out a consistent way to get message bounds across all message view holders
+     */
+    private fun setupMessageView() {
+        messageView = messageViewHolderFactory
+            .createViewHolder(
+                binding.messageContainer,
+                MessageListItemViewTypeMapper.getViewTypeValue(messageItem)
+            )
+            .cast<MessagePlainTextViewHolder>()
+            .apply {
+                addMessageView(itemView)
+                bind(messageItem)
+            }.binding.messageText
+    }
+
+    private fun setupUserReactionsView() {
+        binding.userReactionsView.setMessage(messageItem.message)
+    }
+
+    private fun anchorReactionsViewToMessageView() {
+        val reactionsWidth = requireContext().getDimension(R.dimen.stream_ui_edit_reactions_total_width)
+        val reactionsOffset = requireContext().getDimension(R.dimen.stream_ui_edit_reactions_horizontal_offset)
+
+        messageView.addOnLayoutChangeListener { _, left, _, right, _, _, _, _, _ ->
+            with(binding) {
+                val maxTranslation = messageContainer.width / 2 - reactionsWidth / 2
+                editReactionsView.translationX = if (messageItem.isMine) {
+                    left - messageContainer.width / 2 - reactionsOffset
+                } else {
+                    right - messageContainer.width / 2 + reactionsOffset
+                }.coerceIn(-maxTranslation, maxTranslation).toFloat()
+            }
+        }
+    }
+
+    private fun addMessageView(messageView: View) {
+        binding.messageContainer.addView(
+            messageView,
+            FrameLayout.LayoutParams(
+                FrameLayout.LayoutParams.MATCH_PARENT,
+                FrameLayout.LayoutParams.WRAP_CONTENT
+            )
+        )
+    }
+
+    private class MessageItemWrapper(val messageListItem: MessageListItem.MessageItem) : Serializable
```

Comment:
Why do we need this wrapper class?
Can't we use `MessageListItem.MessageItem` instead?

Reply:
Yes, I will make `MessageItem` serializable then

Reply:
Could we use `Parcelable` instead? Much more reliable in my experience. Also, `MessageItem` contains things like the network `Message` model, I'd be terrified of serializing it. Maybe we could introduce a new, simple model here that has just the necessary data, and pass that through to the dialog instead?

Reply:
Because the Dialog internally is using our ViewModelFacory, we need to use `MessageItem` with its real content.
About `Parcelable`? Why? It doesn't contribute anything. Serializable is easier for this case.

Reply:
Serializable is a reflection based Java API, which means it's a lot slower (doesn't matter in this case), and will not necessarily play nicely with Kotlin. I wouldn't be surprised if something unexpected happened while serializing and deserializing values here.

Reply:
Did it by making `MessageItem` serializable